src/components/Common/Modal/Modal.tsx

```diff
@@ -39,22 +41,31 @@ const Modal = forwardRef<HTMLDivElement, ModalProps>(
     },
     ref
   ) => {
+    // modal이 떠 있을 땐 스크롤 막음
+    // modal 닫히면 다시 스크롤 가능하도록 함
+    useEffect(() => {
+      document.body.style.overflow = isOpen ? 'hidden' : '';
+      document.documentElement.style.height = isOpen ? '100vh' : '';
+    }, [isOpen]);
+
     return createPortal(
       isOpen ? (
         <ModalWrapper ref={ref}>
-          <ModalBackground onClick={onClose} />
+          <ModalBackground onClick={mode === 'normal' ? onClose : undefined} />
           <ModalContainer
             width={width}
             height={height}
             $borderRadius={borderRadius}
             {...props}
           >
-            <ModalCloseButton
-              $coordinate={borderRadius}
-              onClick={onClose}
-            >
-              <CloseRoundedIcon />
-            </ModalCloseButton>
+            {mode === 'normal' ? (
+              <ModalCloseButton
+                $coordinate={borderRadius}
+                onClick={onClose}
+              >
+                <CloseRoundedIcon />
+              </ModalCloseButton>
+            ) : null}
```

Comment:
사소한 부분이지만 null을 반환해주는 값들은
삼항 연산을 && 연산으로 대체해도 좋을 것 같아요!

```suggestion
            {mode === 'normal' && (
              <ModalCloseButton
                $coordinate={borderRadius}
                onClick={onClose}
              >
                <CloseRoundedIcon />
              </ModalCloseButton>
            )}
```